# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/remoteworkers/RemoteWorkerChild.cpp
# Commit: 19c4dc203724
# Full Hash: 19c4dc2037240742eaffc0fa8ae56f6e338a48d4
# Author: Yaron Tausky <ytausky@mozilla.com>
# Date: 2019-03-06 04:36:55
# Regressor Bug: 1530223
# File Overlap Count: 1
# Description:
#   Bug 1530223: Promote assertion to MOZ_RELEASE_ASSERT r=asuth
#   
#   Instead of trying to figure out which pointer is nullptr, add a more
#   powerful assertion. Since the nullptr already causes a crash, this won't
#   cause any regression.
# ==============================================================================

diff -r 2237ab9382ec -r 19c4dc203724 dom/workers/remoteworkers/RemoteWorkerChild.cpp
--- a/dom/workers/remoteworkers/RemoteWorkerChild.cpp	Tue Mar 05 03:37:00 2019 +0000
+++ b/dom/workers/remoteworkers/RemoteWorkerChild.cpp	Tue Mar 05 16:36:24 2019 +0000
@@ -441,7 +441,7 @@
 
   // The holder will be notified by this.
   if (mWorkerState == eRunning) {
-    MOZ_ASSERT(mWorkerPrivate);
+    MOZ_RELEASE_ASSERT(mWorkerPrivate);
     mWorkerPrivate->Cancel();
   }
 }
